app/src/test/java/org/simple/clinic/newentry/PatientEntryScreenControllerTest.kt

```diff
@@ -223,6 +229,39 @@ class PatientEntryScreenControllerTest {
     verify(screen, times(4)).showEmptyStateError(true)
   }
 
+  @Test
+  fun `when input validation errors are emitted, they must be sent to analytics`() {
+    uiEvents.onNext(PatientFullNameTextChanged(""))
+    uiEvents.onNext(PatientNoPhoneNumberToggled(noneSelected = false))
+    uiEvents.onNext(PatientPhoneNumberTextChanged(""))
+    uiEvents.onNext(PatientDateOfBirthTextChanged(""))
+    uiEvents.onNext(PatientAgeTextChanged(""))
+    uiEvents.onNext(PatientGenderChanged(None))
+    uiEvents.onNext(PatientColonyOrVillageTextChanged(""))
+    uiEvents.onNext(PatientNoColonyOrVillageToggled(noneSelected = false))
+    uiEvents.onNext(PatientDistrictTextChanged(""))
+    uiEvents.onNext(PatientStateTextChanged(""))
+    uiEvents.onNext(PatientEntrySaveClicked())
+
+    whenever(dobValidator.validate("33/33/3333")).thenReturn(Result.DATE_IS_IN_FUTURE)
+    uiEvents.onNext(PatientDateOfBirthTextChanged("33/33/3333"))
+    uiEvents.onNext(PatientEntrySaveClicked())
+
+    whenever(dobValidator.validate(" ")).thenReturn(Result.INVALID_PATTERN)
+    uiEvents.onNext(PatientAgeTextChanged(" "))
+    uiEvents.onNext(PatientDateOfBirthTextChanged(""))
+    uiEvents.onNext(PatientEntrySaveClicked())
+
+    whenever(dobValidator.validate("16/07/2018")).thenReturn(Result.INVALID_PATTERN)
+    uiEvents.onNext(PatientDateOfBirthTextChanged("16/07/2018"))
+    uiEvents.onNext(PatientEntrySaveClicked())
+
+    val validationErrors = reporter.receivedEvents
+        .filter { it.name == "InputValidationError" }
+
+    Truth.assertThat(validationErrors).isNotEmpty()
```

Comment:
Can you statically import `Truth` functions? Let's do this change in a future commit.

Reply:
Can we check that the number of reported errors is `X` instead of any positive value? Is that a useful assertion?

Reply:
I don't think it's a useful assertion, because it makes the test fragile. If we add new kinds of errors, or decide to merge some errors internally in the validator, the test will break.

What will be useful if we have a way to access just the stream of errors, so we can assert that the total number of reported events is equal to the number of errors. Does that make sense?